internal/controller/server/execution/execution.go

```diff
@@ -183,27 +182,23 @@ func getActiveExecutorMap(e *execution) *activeExecutorMap {
 	return e.activeExecutorMap
 }
 
-//ExecuteJob function will call job repository and get jobId
-func (e *execution) ExecuteJob(ctx context.Context, jobName string, jobData map[string]string) (uint64, error) {
-	jobAvailabilityStatus, err := e.jobRepo.CheckJobMetadataIsAvailable(ctx, jobName)
+// ExecuteJob function will call job repository and get jobId
+func (e *execution) ExecuteJob(ctx context.Context, jobContext *clientCPproto.RequestForExecute) (uint64, error) {
+	jobAvailabilityStatus, err := e.jobRepo.CheckJobMetadataIsAvailable(ctx, jobContext.JobName)
```

Comment:
This operation is more specific to the metadata repository instead of jobRepo. Can you create a Get method in metadata repo and use that for this query instead of doing it through the jobRepo?

Reply:
Actually, That function is not metadata related.It is job Related. we will change checkjobMetadataIsAvailable to checkJobAvailable.